src/pages/components/map-related/MapContent.tsx

```diff
@@ -75,11 +80,13 @@ export function MapContent({ positions, plannedRouteStops, showNavigationButtons
           imgSrc={actualRouteStopMarkerPath}
           title={t('actualRoute')}
         />
-        <MapIndex
-          lineColor={plannedRouteLineColor}
-          imgSrc={plannedRouteStopMarkerPath}
-          title={t('plannedRoute')}
-        />
+        {showPlannedRoute && (
```

Comment:
Did you consider this option?

Can be cleaner :)

```suggestion
        {plannedRouteStops && (
```

Reply:
You're absolutely right – it's much cleaner this way.
Updated the code accordingly. Thanks for the suggestion!